File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        s = list(s)
        l = set('aeiouAEIOU')
        i, j = 0, len(s) - 1
        while i < j:
            while s[i] not in l and i < j:
                i += 1
            while s[j] not in l and i < j:
                j -= 1
            s[i], s[j] = s[j], s[i]
            i += 1
            j -= 1
        return "".join(s)

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        char[] result = s.toCharArray();
        String vowels = "aeiouAEIOU";
        int left=0; int right = result.length -1;
            while (left < right) {

            while (left < right && vowels.indexOf(result[left]) == -1) {
                left++;
            }
            while (left < right && vowels.indexOf(result[right]) == -1) {
                right--;
            }
            char temp = result[left];
            result[left] = result[right];
            result[right] = temp;

            left++;
            right--;
        }
        return new String(result);
        
    }
}